tensorflow/core/kernels/scatter_nd_op.cc

```diff
@@ -165,6 +165,11 @@ class TensorScatterOp : public OpKernel {
                                       updates.shape().num_elements()),
                 errors::InvalidArgument(
                     "Indices and updates specified for empty output shape"));
+    OP_REQUIRES(c, 
+                !(updates.shape().dims() < indices.shape().dims() - 1),
```

Comment:
```suggestion
                updates.shape().dims() >= indices.shape().dims() - 1,
```

This is more readable

Reply:
The full list of condition is listed [here](https://www.tensorflow.org/api_docs/python/tf/tensor_scatter_nd_update#in_general), with comments indicating the values from the reported bug:
```python
assert tf.rank(indices) >= 2                                     # assert 3 >= 2
index_depth = indices.shape[-1]                            # 1
batch_shape = indices.shape[:-1]                          # [1, 1]
assert index_depth <= tf.rank(tensor)                   # 1 <= 1
outer_shape = tensor.shape[:index_depth]          # [2]
inner_shape = tensor.shape[index_depth:]          # []
assert updates.shape == batch_shape + inner_shape   # assert [1, 1] == [1]
```
So that last condition fails, and is the missing check.

Reply:
Technically this Op is working when we pass updates even higher rank than indices. Please refer attached [gist](https://colab.sandbox.google.com/gist/SuryanarayanaY/ad3e0541f514f48066f9f82889707163/63375_r2.ipynb).

Reply:
Yes, it can be higher.  If you copy-paste those above conditions and asserts, you will find they all pass for all your examples.

Reply:
No, it is _not_ okay.  The assert that should fail is the last one I pasted above, which has an _equality_ assertion on the sizes of `updates.shape == batch_shape + inner_shape`.

Reply:
Hi @SuryanarayanaY Any update on this PR? Please. Thank you!

Reply:
Hi @SuryanarayanaY Any update on this PR? Please. Thank you!